src/app-layout/__tests__/runtime-drawers.test.tsx

```diff
@@ -633,3 +648,418 @@ describeEachAppLayout(({ theme, size }) => {
+        ...drawerDefaults,
+        id: 'global-drawer-2',
+        type: 'global',
+        defaultActive: true,
+        mountContent: container => (container.textContent = 'global drawer content 2'),
+      });
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'global-drawer-3',
+        type: 'global',
+        mountContent: container => (container.textContent = 'global drawer content 3'),
+      });
+      const { wrapper } = await renderComponent(<AppLayout drawers={[testDrawer]} />);
+
+      expect(wrapper.findActiveDrawers()!.length).toBe(2);
+      expect(wrapper.findActiveDrawers()[0].getElement()).toHaveTextContent('global drawer content 1');
+      expect(wrapper.findActiveDrawers()[1].getElement()).toHaveTextContent('global drawer content 2');
+
+      wrapper.findDrawerTriggerById('global-drawer-3')!.click();
+
+      expect(wrapper.findActiveDrawers()!.length).toBe(2);
+      expect(wrapper.findActiveDrawers()[0].getElement()).toHaveTextContent('global drawer content 2');
+      expect(wrapper.findActiveDrawers()[1].getElement()).toHaveTextContent('global drawer content 3');
+    });
+
+    test('first opened drawer (global drawer) should be closed when active drawers take up all available space on the page and a third drawer is opened', async () => {
+      jest.mocked(getLogicalBoundingClientRect).mockReturnValue({
+        blockSize: 1291,
+        inlineSize: 1400,
+        insetBlockStart: 45,
+        insetBlockEnd: 1336,
+        insetInlineStart: 0,
+        insetInlineEnd: 1400,
+      });
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'local-drawer',
+        mountContent: container => (container.textContent = 'local-drawer content'),
+      });
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'global-drawer-1',
+        type: 'global',
+        defaultActive: true,
+        mountContent: container => (container.textContent = 'global drawer content 1'),
+      });
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'global-drawer-2',
+        type: 'global',
+        defaultActive: true,
+        mountContent: container => (container.textContent = 'global drawer content 2'),
+      });
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'global-drawer-3',
+        mountContent: container => (container.textContent = 'global drawer content 3'),
+      });
+      const { wrapper } = await renderComponent(<AppLayout drawers={[testDrawer]} />);
+
+      expect(wrapper.findActiveDrawers()!.length).toBe(2);
+      expect(wrapper.findActiveDrawers()[0].getElement()).toHaveTextContent('global drawer content 1');
+      expect(wrapper.findActiveDrawers()[1].getElement()).toHaveTextContent('global drawer content 2');
+
+      await delay();
+
+      wrapper.findDrawerTriggerById('local-drawer')!.click();
+
+      await delay();
```

Comment:
Because active drawers don't get updated immediately after a user's action. Replaced it with `waitFor` as it's more "natural" way to handle such conditions